### PR TITLE
Set up zone room

### LINKED ISSUE #45: D3_Set up Zone and Room in SHH
add zone to SHH
add room to the zone 
save the zone room map to the house Room model for the application

### LINKED ISSUE #47: D3_set Morning Day and night Temperature
set Morning temperature for specific zone 
set Day temperature for specific zone
set Night temperature for specific zone
update room temperature of the same zone

### PR DESCRIPTION
**Pull request by kimchhengheng**
create zone, delete zone
add room to existing zone

### CODE REVIEW

**Review by foreversponge — APPROVED** [review]

**Review by BadrSaidi — APPROVED** [review]

File: logs.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Nov 26, 2020 9:11:40 PM models.LogToFileModel log
```
Comment (d-rossi) [inline logs.txt]: Do not commit the logs.txt file

File: src/main/java/controllers/RoomController.java [hunk 2]
```diff
@@ -40,6 +38,8 @@
 	private HouseRoomsModel houseRoomsModel= HouseRoomsModel.getInstance();
 	private RoomModel room;
 
+
+
```
Comment (d-rossi) [inline src/main/java/controllers/RoomController.java]: Remove useless whitespace

File: src/main/java/controllers/SHHController.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import javafx.event.ActionEvent;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import models.HouseRoomsModel;
+import models.RoomModel;
+
+import java.io.IOException;
+import java.util.Set;
+
+public class SHHController {
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java:21]: Add javadoc for the class

File: src/main/java/controllers/SHHController.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+     * each room have an instance of the Temperature so change the new value with the existing one
+     * @param event
+     */
+    public void setMorningTemperature(ActionEvent event) {
+        temperatureInput.clear();
+        if(!checkSelectZone() || !checkValue()){
+            return ;
+        }
+        String selectZone= (String) zoneRoomList.getSelectionModel().getSelectedItem();
+        double inputT = Double.parseDouble(temperatureInput.getText());
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java]: fix typo or provide better naming for "inputT".

File: src/main/java/controllers/SHHController.java [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+     * each room have an instance of the Temperature so change the new value with the existing one
+     * @param event
+     */
+    public void setDayTemperature(ActionEvent event) {
+        temperatureInput.clear();
+        if(!checkSelectZone() || !checkValue()){
+            return ;
+        }
+        String selectZone= (String) zoneRoomList.getSelectionModel().getSelectedItem();
+        double inputT = Double.parseDouble(temperatureInput.getText());
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java]: fix typo or provide better naming for "inputT".

File: src/main/java/controllers/SHHController.java [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+     * each room have an instance of the Temperature so change the new value with the existing one
+     * @param event
+     */
+    public void setNightTemperature(ActionEvent event) {
+        temperatureInput.clear();
+        if(!checkSelectZone() || !checkValue()){
+            return ;
+        }
+        String selectZone= (String) zoneRoomList.getSelectionModel().getSelectedItem();
+        double inputT = Double.parseDouble(temperatureInput.getText());
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java]: fix typo or provide better naming for "inputT".

File: src/main/java/controllers/SHHController.java [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        Set<String> roomInZone = houseRoomsModel.getZoneRoomMap().get(zoneRoomArray[0]);
+        for(String s: roomInZone){
+            for(RoomModel rm : roomModels){
+                if(s.equalsIgnoreCase(rm.getName())){
+                    rm.getTemperature().setMorningTemp(inputT);
+                }
+            }
+        }
+    }
+    public Boolean checkSelectZone(){
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java:134]: Add javadoc for this method

File: src/main/java/controllers/SHHController.java [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+        String selectZone= (String) zoneRoomList.getSelectionModel().getSelectedItem();
+        if(selectZone == null){
+            errorLabel.setText("* please select the zone");
+            return false;
+        }
+        return true;
+    }
+
+    public Boolean checkValue(){
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java:149]: Add javadoc for this method

File: src/main/java/controllers/SHHController.java [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+        String tempValue = temperatureInput.getText();
+        boolean isNumber=false;
+        double inputT;
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java]: fix typo or provide better naming for "inputT".

File: src/main/java/controllers/SHHController.java [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        try{
+            inputT = Double.parseDouble(tempValue);
+            isNumber =true;
+        }
+        catch (NumberFormatException e){
+            errorLabel.setText("* input temperature have to be number");
+
```
Comment (d-rossi) [inline src/main/java/controllers/SHHController.java]: Remove useless whitespace

File: src/main/java/controllers/dashBoardController.java [hunk 11]
```diff
@@ -58,29 +56,102 @@
 	private String selectItem;
 	private List<String> selectLocation;
 
+
+	public Map<String, ArrayList<LocalTime>> getAwayModeLight() {
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:67]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 12]
```diff
@@ -58,29 +56,102 @@
...
+		return awayModeLight;
+	}
+
+	public void setAwayModeLight(Map<String, ArrayList<LocalTime>> awayModeLight) {
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:74]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 13]
```diff
@@ -58,29 +56,102 @@
...
+		this.awayModeLight = awayModeLight;
+	}
+
+	private Map<String, ArrayList<LocalTime>> awayModeLight= new HashMap<>();
+
 	@FXML private ScrollPane scroll;
 	@FXML private GridPane grid;
 
+//	public void HandleSHH(Event event) {
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java]: Remove commented it out lines of code. (lines 73 to 82)

File: src/main/java/controllers/dashBoardController.java [hunk 14]
```diff
@@ -58,29 +56,102 @@
...
 	 */
 	class IncrementTask extends TimerTask{
 		private LocalTime localTime;
 		private int timeInc =1;
+		private String mode;
+		public IncrementTask(){
+			this.mode = "Increment";
+		}
+
+		public IncrementTask(String mode){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:96]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 15]
```diff
@@ -58,29 +56,102 @@
...
+			this.mode = mode;
+		}
+
 		public void setTimeInc(int timeInc) {
 			this.timeInc = timeInc;
 		}
 		private void setTime(LocalTime ctime) {
 			this.localTime = ctime;
 		}
+		private void awayModeLight(){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:119]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 16]
```diff
@@ -58,29 +56,102 @@
...
+			boolean update =false;
+			if(!awayModeLight.isEmpty()) {
+//				System.out.println(houseRoomsModel.getAllRoomsArray());
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java]: Remove commented out line of code

File: src/main/java/controllers/dashBoardController.java [hunk 17]
```diff
@@ -58,29 +56,102 @@
...
+							}
+						}
+					}
+				}
+			}
+			if(update){
+				displayLayout();
+			}
+		}
+		private void increment(){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:150]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 18]
```diff
@@ -58,29 +56,102 @@
...
+			localTime = localTime.plusSeconds(timeInc);
+			time.setText(localTime.format(DateTimeFormatter.ofPattern("HH:mm:ss")));
+			awayModeLight();
+		}
+		private void decrement(){
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:159]: Add javadoc for this method

File: src/main/java/controllers/dashBoardController.java [hunk 19]
```diff
@@ -58,29 +56,102 @@
...
+			localTime = localTime.minusSeconds(timeInc);
+
```
Comment (d-rossi) [inline src/main/java/controllers/dashBoardController.java:161]: Remove useless whitespace

File: src/main/java/controllers/setUpZoneRoomController.java [hunk 20]
```diff
@@ -0,0 +1,142 @@
...
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.stage.Stage;
+import models.*;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class setUpZoneRoomController {
```
Comment (d-rossi) [inline src/main/java/controllers/setUpZoneRoomController.java:20]: Add javadoc for this class

File: src/main/java/models/Temperature.java [hunk 21]
```diff
@@ -0,0 +1,108 @@
+package models;
+
+public class Temperature {
```
Comment (d-rossi) [inline src/main/java/models/Temperature.java:8]: Add javadoc for this class

File: logs.txt (hunk 1 above)
Comment (kimchhengheng) [reply logs.txt]: it is from previous delivery but i delete now

File: src/main/java/controllers/SHHController.java (hunk 4 above)
Comment (kimchhengheng) [reply src/main/java/controllers/SHHController.java]: it not type it is inputTemperature but it would too long

**Pull request merged**